internal/dataexchange/ffdx/ffdx.go

```diff
@@ -67,11 +74,17 @@ func (cb *callbacks) OperationUpdate(ctx context.Context, update *core.Operation
 	}
 }
 
-func (cb *callbacks) DXEvent(ctx context.Context, namespace string, event dataexchange.DXEvent) {
-	if handler, ok := cb.handlers[namespace]; ok {
-		handler.DXEvent(cb.plugin, event)
+func (cb *callbacks) DXEvent(ctx context.Context, namespace, recipient string, event dataexchange.DXEvent) {
+	if node, ok := cb.plugin.nodes[recipient]; ok {
```

Comment:
Is it possible for a DX message to arrive _before_ the node identity has been confirmed? Here `plugin.nodes` is populated with any nodes that exist on startup or any nodes that are confirmed by an identity claim broadcast. I don't see any way to pick the correct handler unless the node has been claimed and confirmed.

Reply:
Basically we have a validation on the new "recipient" field provided by DX, to see that it matches a known node/verifier in the database. Previously this field didn't exist, so any message received was just put into the database.

Reply:
So the question is if you were to:
1. As `B` broadcast my node to the network
2. As `A` triggers an event off of the broadcast of `B` and _immediately_ send a message to a group containing `A`
3. As `A` receive the message from `B` _before_ processing the broadcast from (1) - even though it was successfully processed by the remote node.

I believe it is (at least theoretically) possible to engineer this scenario.
It doesn't seem like one to worry about in practice.

Reply:
Yea, I think it would mainly be if the blockchain listener on `B` was disconnected or behind when (3) is received. It would ignore the off-chain data since there's no orchestrator lined up to handle it until (1) is confirmed (on- and off-chain portions) on my local node. It feels like an edge case though.

Perhaps the more important angle to this is that there's no way to communicate back to `A` whether the message was processed. DX will indicate success as long as the message was delivered to FireFly - but if FireFly decides to drop it on the floor, `A` will never know.

Reply:
In the more general case, you could send a message to `peer0/jack` and DX would report success as long as `peer0` is valid - regardless of whether `/jack` exists at all.

Reply:
spoke to @gabriel-indik about this, and we agreed the MTLS implementation of DX should reject a HTTP POST to an unregistered node ID

Reply:
OK - can table it in the context of this PR then, and come back after scoping what it would take to push that responsibility into DX